cunumeric/array.py

```diff
@@ -823,6 +823,21 @@ def __ge__(self, rhs):
 
     def _convert_key(self, key, first=True):
         # Convert any arrays stored in a key to a cuNumeric array
+        if isinstance(key, slice):
+            if isinstance(key.start, ndarray):
```

Comment:
I'd suggest you do the integer conversion unconditionally.  An `isinstance` call isn't particularly cheaper.